### PR TITLE
Simplify the work with generated BoundNodes and emit the xml comments as doc comment in the generated code file

### PR DESCRIPTION
**Pull request by bernd5**
The "BoundNodes.xml" contains a lot of very useful comments for working with those nodes.

This pull request now adds those comments to the generated code.

### CODE REVIEW

**Comment by bernd5** [comment]
@CyrusNajmabadi can you have a look? What do you think?

File: src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/Deserializer.cs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+using System.Reflection;
+using System.Xml.Serialization;
+using System.Xml.Linq;
+using System.Collections;
+using System.IO;
+using System.Threading;
+
+namespace BoundTreeGenerator
+{
+    internal class Deserializer
```
Comment (333fred) [inline src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/Deserializer.cs:16]: It feels like, rather than implementing a custom deserializer and comment handler, it would be much simpler to just turn those comments into XML nodes, like we do for IOperation `<summary>` nodes. It does require updates to the xml doc to convert the existing comments to this new form, but it feels more maintainable long-term.

**Comment by AlekseyTs** [comment]
>   <ValueType Name="UnaryOperatorKind"/>

In my opinion, there are too many changes in this file. Given the stated goal of the PR "The "BoundNodes.xml" contains a lot of very useful comments for working with those nodes.This pull request now adds those comments to the generated code.", I expect to see no changes in this file at all. I suggest reverting any formatting changes in this file, so that reviewers could clearly see the meaningful changes, if any. 

---
Refers to: src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml:28 in fd78eb6. [](commit_id = fd78eb649894d5e2e01bea2d73f00aa6f99e3d51, deletion_comment = True)

**Comment by AlekseyTs** [comment]
>     <ValueType Name="QualificationKind"/>

Similar comment. Please revert any stylistic/formatting changes

---
Refers to: src/Compilers/VisualBasic/Portable/BoundTree/BoundNodes.xml:27 in fd78eb6. [](commit_id = fd78eb649894d5e2e01bea2d73f00aa6f99e3d51, deletion_comment = True)

File: src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/Model.cs [hunk 2]
```diff
@@ -33,9 +59,21 @@ public class TreeType
...
+            return ref Comments;
+        }
+    }
+
+    public sealed class PredefinedNode : TreeType
+    {
     }
 
-    public class PredefinedNode : TreeType
+    public sealed class ValueType : TreeType
```
Comment (AlekseyTs) [inline src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/Model.cs:76]: >public sealed class ValueType : TreeType [](http://example.com/codeflow?start=4&length=40)

Consider leaving the type where it used to be. That will make the nature of the change clearer, thus it will be easier to follow it

File: src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/Program.cs [hunk 3]
```diff
@@ -6,52 +6,33 @@
...
             if (args.Length != 3)
             {
                 Console.Error.WriteLine("Usage: \"{0} <language> <input> <output>\", where <language> is \"VB\" or \"CSharp\"", Path.GetFileNameWithoutExtension(args[0]));
                 return 1;
             }
 
-            language = args[0];
-            infilename = args[1];
-            outfilename = args[2];
+            var language = args[0];
```
Comment (AlekseyTs) [inline src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/Program.cs:25]: >var language = args[0]; [](http://example.com/codeflow?start=12&length=23)

Please revert style changes for the existing code.

**Comment by bernd5** [comment]
> > 
> 
> I expect to see no changes in this file at all. I suggest reverting any formatting changes in this file, so that reviewers could clearly see the meaningful changes, if any.
> 
> Refers to: src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml:28 in [fd78eb6](https://github.com/dotnet/roslyn/commit/fd78eb649894d5e2e01bea2d73f00aa6f99e3d51). [](commit_id = [fd78eb6](https://github.com/dotnet/roslyn/commit/fd78eb649894d5e2e01bea2d73f00aa6f99e3d51), deletion_comment = True)

This was what I did first - I actually wanted no change in the xml file. But Fred Silberberg said that it we better to model doc comments as actual xml nodes.

I generated the xml file via XmlSerializer - not sure how to handle the diff.
I tried to reduce the diff...

File: src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml [hunk 4]
```diff
@@ -38,204 +38,230 @@
   <ValueType Name="InterpolatedStringHandlerData"/>
   <ValueType Name="WellKnownMember"/>
   <ValueType Name="CollectionExpressionTypeKind"/>
-
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml:41]: My suggestion would be to manually go through this diff and revert the unnecessary changes like these blank line removals. I understand that it's a bit of work to do, but I think that it will produce a better diff and that the XML-first approach is better for the long-term health of the code.

**Comment by jaredpar** [comment]
Closing as this PR has gotten stale. Can re-open if there is interest in moving it forward.
